packages/app/non-translatable-markup/src/methods/utils.ts

```diff
@@ -1,8 +1,28 @@
 import { NON_TRANSLATABLE_END_TAG, NON_TRANSLATABLE_START_TAG } from '../nonTranslatableTags'
 
 const nonTranslatableTextPattern = `${NON_TRANSLATABLE_START_TAG}.*?${NON_TRANSLATABLE_END_TAG}`
-const nonTranslatableTagsPattern = `[${NON_TRANSLATABLE_START_TAG}${NON_TRANSLATABLE_END_TAG}]`
-
 export const nonTranslatableTextRegexp = new RegExp(nonTranslatableTextPattern)
 export const nonTranslatableTextRegexpG = new RegExp(nonTranslatableTextPattern, 'g')
+
+const nonTranslatableTagsPattern = `[${NON_TRANSLATABLE_START_TAG}${NON_TRANSLATABLE_END_TAG}]`
 export const nonTranslatableTagsRegexpG = new RegExp(nonTranslatableTagsPattern, 'g')
+
+/**
+ * Explanation of the pattern:
+ *  `[...]` -> matches any one of the characters or Unicode property escapes specified within it
+ *      `\p{P}` -> matches any kind of punctuation character
+ *      `\p{Sm}` -> matches any math symbol character
+ *      `\p{Sk}` -> matches any modifier symbol character
+ *      `\p{So}` -> matches any other symbol character
+ *      `\p{Emoji}` -> matches any emoji character
+ *  `u` -> enables unicode mode
+ *
+ * Note: `\p{S}` matches any kind of symbol character, and it covers all the symbols from `\p{Sm}`, `\p{Sc}`, `\p{Sk}`,
+ *  and `\p{So}` categories. However, we don't want to cover `\p{Sc}` (Currency Symbol), so we specify the relevant
+ *  subcategories separately.
+ *
+ * For more info see -> https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/regular_expressions/unicode_character_class_escape/index.md
+ */
+const symbolsPattern = /[\p{P}\p{Sm}\p{Sk}\p{So}\p{Emoji}']/u
```

Comment:
If I see correctly, for checking symbols you split text by symbols and then check each part recursively? Won't it work if we use:
```
/[\p{P}\p{Sm}\p{Sk}\p{So}\p{Emoji}']+/u
```
And then check if whole string is matching regex?

Reply:
Fixed, thank youuu